src/wsproxy.ml

```diff
@@ -0,0 +1,108 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+
+let get_dir_path () = Printf.sprintf "/var/xapi/" 
+
+
+module LwtWsIteratee = Websockets.Wsprotocol(Lwt)
+
+
+let start path handler =
+  let dir_path = get_dir_path () in
+  let fd_sock_path = Printf.sprintf "%s%s" dir_path path in
+  let fd_sock = Lwt_unix.socket Unix.PF_UNIX Unix.SOCK_STREAM 0 in
+  let%lwt () = (try%lwt Lwt_unix.unlink fd_sock_path with _ -> Lwt.return ()) in
+  let () = Lwt_unix.Versioned.bind_1 fd_sock (Unix.ADDR_UNIX fd_sock_path) in
+  let () = Lwt_unix.listen fd_sock 5 in
+
+  let rec loop () =
+    try%lwt
+      let%lwt (fd_sock2,_) = Lwt_unix.accept fd_sock in
+      let buffer = String.make 16384 '\000' in
+      let iov = Lwt_unix.io_vector buffer 0 16384 in
+      let%lwt (len,newfds) = 
+        try
+          Lwt_unix.recv_msg fd_sock2 [iov] 
+        with e ->
+          let%lwt () = Lwt_unix.close fd_sock2 in
+          Lwt.return (0,[])
+      in
+      let msg = String.sub buffer 0 len in
+      let%lwt _ = Lwt_unix.close fd_sock2 in
+      List.iter (fun fd -> Printf.printf "got fd: %d\n%!" (Obj.magic fd)) newfds;
+      Printf.printf "About to fixup the fd\n%!";
+      ignore(handler (Lwt_unix.of_unix_file_descr (List.hd newfds)) msg);
+      loop ()
+    with e ->
+      Printf.printf "Caught exception: %s\n" (Printexc.to_string e);
+      loop ()
+  in
+  loop ()
+
+let ($) f x = f x
```

Comment:
This is a Haskell-ism but it OCaml it associates to the left, which is wrong. That is why `@@` is used for this in OCaml, because it associates to the right. I also don't see it being used.